stock-service/src/main/java/com/ecommercepractice/stockservice/controller/StockController.java

```diff
@@ -0,0 +1,107 @@
+    }
+
+    @ApiOperation(value ="FETCH BY ID", notes = "Get a product on stock by their id.")
+    @GetMapping("/products/{stockId}")
+    public ResponseEntity<EntityModel<Stock>> fetchByStockId(@PathVariable(required = true) Long stockId) {
+        log.info(String.format("STOCK | FETCH | PAYLOAD { StockId -> %s } ", stockId));
+        return new ResponseEntity(stockModelAssembler.toModel(stockService.fetchByStockId(stockId)), HttpStatus.OK);
+    }
+
+    @ApiOperation(value ="UPDATE BY ID", notes = "Update stockQuantity and/or salePrice on stock by their id.")
+    @PutMapping("products/{stockId}")
+    public ResponseEntity<EntityModel<Stock>> updateByStockId(@PathVariable(required = true) Long stockId,
+                                                 @RequestParam(required = false) Long stockQuantity,
+                                                 @RequestParam(required = false) Float salePrice
+    ) {
+        log.info(String.format("STOCK | UPDATE | PAYLOAD { StockId -> %s , Stock -> %s } ", stockId,
+                stockQuantity == null ? "NO_QUANTITY" : stockQuantity,
+                salePrice == null ? "NO_SALE_PRICE" : salePrice
+        ));
+        return new ResponseEntity(stockModelAssembler.toModel(stockService.updateByStockId(stockId, stockQuantity, salePrice)), HttpStatus.OK);
+    }
+    @ApiOperation(value ="DELETE BY ID", notes = "DELETE a product on stock by their id.")
+    @DeleteMapping("products/{stockId}")
+    public ResponseEntity deleteByStockId(@PathVariable(required = true) Long stockId) {
+        log.info(String.format("STOCK | DELETE | PAYLOAD { StockId -> %s } ", stockId));
+        stockService.deleteByStockId(stockId);
+        return new ResponseEntity(HttpStatus.NO_CONTENT);
+    }
+
+    @ApiOperation(value ="FREEZE BY ID", notes = "FREEZE a product on stock by their id. When a product is freezed means" +
+            "that is inside a car.")
+    @PostMapping("products/{stockId}/freeze/{freezeQuantities}")
+    public ResponseEntity freezeStockItem(@PathVariable(required = true) Long stockId, @PathVariable(required = true) Long freezeQuantities) {
+        log.info(String.format("STOCK | FREEZE | PAYLOAD { StockId -> %s} ", stockId));
+        stockService.freezeStockItem(stockId, freezeQuantities);
+        return new ResponseEntity(HttpStatus.NO_CONTENT);
+    }
+
+    @ApiOperation(value ="BOUGHT BY ID", notes = "BOUGHT a product on stock by their id. That means, the product is" +
+            "already paid, and it is not available for being purchased")
+    @PostMapping("products/{stockId}/buy/{QuantityToBuy}")
+    public ResponseEntity boughtStockItem(@PathVariable(required = true) Long stockId, @PathVariable Long QuantityToBuy) {
+        log.info(String.format("STOCK | BOUGHT | PAYLOAD { StockId -> %s } ", stockId));
+        stockService.boughtStockItem(stockId, QuantityToBuy);
+        return new ResponseEntity(HttpStatus.NO_CONTENT);
```

Comment:
If this method does not return a body try to change to Response only

Reply:
I don't understand, what kind of object should the method respond? I only found Response in Java ws rs core package I don't consider that appropriate.